docs/proposals/20210310-edge-device-management.md

```diff
@@ -0,0 +1,498 @@
+	ValueTypeUint16       PropertyValueType = "Uint16"
+	ValueTypeUint32       PropertyValueType = "Uint32"
+	ValueTypeUint64       PropertyValueType = "Uint64"
+	ValueTypeInt8         PropertyValueType = "Int8"
+	ValueTypeInt16        PropertyValueType = "Int16"
+	ValueTypeInt32        PropertyValueType = "Int32"
+	ValueTypeInt64        PropertyValueType = "Int64"
+	ValueTypeFloat32      PropertyValueType = "Float32"
+	ValueTypeFloat64      PropertyValueType = "Float64"
+	ValueTypeBinary       PropertyValueType = "Binary"
+	ValueTypeBoolArray    PropertyValueType = "BoolArray"
+	ValueTypeStringArray  PropertyValueType = "StringArray"
+	ValueTypeUint8Array   PropertyValueType = "Uint8Array"
+	ValueTypeUint16Array  PropertyValueType = "Uint16Array"
+	ValueTypeUint32Array  PropertyValueType = "Uint32Array"
+	ValueTypeUint64Array  PropertyValueType = "Uint64Array"
+	ValueTypeInt8Array    PropertyValueType = "Int8Array"
+	ValueTypeInt16Array   PropertyValueType = "Int16Array"
+	ValueTypeInt32Array   PropertyValueType = "Int32Array"
+	ValueTypeInt64Array   PropertyValueType = "Int64Array"
+	ValueTypeFloat32Array PropertyValueType = "Float32Array"
+	ValueTypeFloat64Array PropertyValueType = "Float64Array"
+)
+
+// Command defines the available commands for end users to control the devices
+// NOTE: a Command usually corresponding to a DeviceCommand and a DeviceResource
+type Command struct {
+    // Command name (unique on the profile)
+    Name string `json:"name,omitempty"`
+    // Get Command
+    Get Get `json:"get,omitempty"`
+    // Put Command
+    Put Put `json:"put,omitempty"`
+}
+
+// Put defines the details of the Put operation
+type Put struct {
+    Path string `json:"path,omitempty"`
+    ParameterNames []string `json:"parameterNames,omitempty"`
+}
+
+// Get defines the details of the Get operation
+type Get struct {
+    Path string `json:"path,omitempty"`
+}
+```
+
+#### 2. Device
+
+Followings are the definition of the `Device` CRD and its related Golang structs:
+
+```go
+// Device is the Schema for the devices API
+type Device struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    
+    Spec   DeviceSpec   `json:"spec,omitempty"`
+    Status DeviceStatus `json:"status,omitempty"`
+}
+
+// DeviceSpec defines the desired state of Device
+type DeviceSpec struct {
+    // Description of the device
+    Description string `json:"description,omitempty"`
+    // Device service specific location 
+    Location string `json:"location,omitempty"`
+    // Associated Device Service - One per device
+    Service string `json:"service"`
+    // Associated Device Profile - Describes the device
+    Profile string `json:"profile"`
+    // DeviceProperties defines the desired states of device properties
+    DeviceProperties map[string]DeviceProperty `json:"deviceProperties,omitempty"`
+}
+
+// DeviceStatus defines the observed state of Device
+type DeviceStatus struct {
```

Comment:
Do we need to consider more device status like the admin status and operation status?

some case like you just install a device but not ready for usage. 

https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/models/adminstate.go
https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/models/operatingstate.go

Those status definition is pretty common in IOT solutions.

Reply:
Would you explain a little bit what is the `operation status` used for?